src/tools/fuzzing.h

```diff
@@ -2115,8 +2115,8 @@ class TranslateToFuzzReader {
       if (heapType.isSignature()) {
         sig = heapType.getSignature();
       } else {
-        assert(heapType == HeapType::func);
-        // The specific signature does not matter.
+        // This is something like funcref or externref, and so the specific
+        // signature does not matter.
```

Comment:
- Isn't `funcref` handled at the very top?
- If this is `externref`, isn't returning of a function an error? It is not a subtype of `externref`..
- Can this also be other data that is not `dataref`?

Reply:
Good point that funcref is handled above. It is a little weird to handle `(ref func)` the non-nullable version separately lower down. This is probably worth refactoring, but for now I think it's good enough.

About dataref, is, `isData` can return true for struct and array types.

Reply:
> About dataref, `isData` can return true for struct and array types also.

I can't see uses of `isData` in this function, no?

By the way I don't mean we should use it here. You added `!type.isFunction()` and I think that covers the data case, in which we can't do much anyway.

Reply:
Oh, sorry, I had `isData()` in an earlier version. Then I switched it to that `!isFunction()` so it also covers `externref`. Yes, I think that's good enough for the data case for now.